testing/demo/pkg/setup/initialise_groth16_client.go

```diff
@@ -59,10 +64,30 @@ func InitializeGroth16LightClientOnSimapp() error {
 }
 
 func createClientAndConsensusState(genesisBlock, latestBlock *ethtypes.Block) (*cdctypes.Any, *cdctypes.Any, error) {
-	// TODO: this clientState isn't created with a state transition verifier key or state inclusion verifier key.
-	// Query the EVM prover for the verifier keys
-	// See: https://github.com/celestiaorg/celestia-zkevm-ibc-demo/blob/main/proto/prover/v1/prover.proto#L16-L20
-	clientState := groth16.NewClientState(latestBlock.Number().Uint64(), []byte{}, []byte{}, []byte{}, genesisBlock.Root().Bytes())
+	// Query the info endpoint for the state transition verifier key
+	// Membership proofs are currently not supported
+	conn, err := grpc.NewClient("localhost:50052", grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to connect to prover: %w", err)
+	}
+	defer conn.Close()
+
+	fmt.Printf("Getting evm prover info...\n")
+	proverClient := proverclient.NewProverClient(conn)
+	info, err := proverClient.Info(context.Background(), &proverclient.InfoRequest{})
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get evm prover info %w", err)
+	}
+	fmt.Printf("Got evm prover info. StateTransitionVerifierKey: %v", info.StateTransitionVerifierKey)
+	verifierKeyDecoded, err := hex.DecodeString(strings.TrimPrefix(info.StateTransitionVerifierKey, "0x"))
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to decode verifier key %w", err)
+	}
+	// Q: Do we still need the fixed size of 32 bytes considering the verifier key is of type []byte?
```

Comment:
@rootulp you'd probably know

Reply:
If this code can pass a `[]byte` to `groth16.NewClientState` then no need to copy the byte slice into a fixed 32 byte array. In other words, let's do what makes Go compiler happy.

If we need to modify types exposed by the groth16 library then I'm fine with both `[]byte` or `[32]byte for the verifier key. `[32]byte` verifier key is more explicit and probably a better practice but not blocking on it if we're not already changing the type.